Exploration Queries/DnsEvents/LeastPrevClientIP-DNSNameQueryToIP.txt

```diff
@@ -1,27 +1,23 @@
-// Name: DNS Name Lookup Query from least prevalent ClientIP to remote IP Address
-// Description: Summary of Bottom 10 Client IP and Domain Names for a given remote IPAddress from DnsEvent Lookup Query data.
-//
-// Entity: IPAddress
-// Input: IPAddress
-// Output: ClientIP
-//
-// QueryPeriod:  +-3h
-//
-// Data Source: #DnsEvents
-//
+// Id: 801bacb0-612a-4195-a84f-7939cca63b92
+// DisplayName: DNS Name Lookup Query from least prevalent ClientIP to remote IP Address
+// Description: Summary of Bottom 10 Client IP and Domain Names for a given remote IPAddress from DnsEvent Lookup Query data (set time range to +-3h when running the query)
+// InputEntityType: Ip
+// InputFields: [Address]
+// OutputEntityTypes: [Ip]
+// QueryPeriodBefore: 3h
+// QueryPeriodAfter: 3h
+// DataSources: #DnsEvents
 // Tactics: #CommandAndControl, #Exfiltration
-//
-let GetAllIPByClientIP = (suspiciousEventTime:datetime, v_IPAddress:string){
-let v_StartTime = suspiciousEventTime-3h;
-let v_EndTime = suspiciousEventTime+3h;
+
+let GetAllIPByClientIP = (v_IP_Address:string){
 DnsEvents
-| where TimeGenerated between (v_StartTime .. v_EndTime)
-| where SubType == "LookupQuery"
-| where IPAddresses contains v_IPAddress
+| where SubType == 'LookupQuery'
+| where IPAddresses contains v_IP_Address
 | summarize min(TimeGenerated), max(TimeGenerated), makeset(Name), count() by ClientIP, IPAddresses
```

Comment:
always make sure to keep all the columns required for the construction of the fields in the output entity types, and **remove the ones that are not used, since this might cause duplicates in the constructed entities when ignoring these extra columns**